### PR TITLE
Enums

### PR DESCRIPTION
**Pull request by danthe1st**
### Request issue
#40 

I cannot self-assign to this issue since I don't have the necessary permissions.
### Website section
`Getting to Know the Language`>`Classes and Objects`>`Enums`

### Details
This PR adds an article on the topic of enums in Java.
It covers the following questions:
- What are enums?
- How can enums be used?
- How do enums differ from normal classes?

Since the `Classes and Objects` section is not present in this repository, it might be necessary to change the directory name, id or similar. For that reason, I have named the file `01_enums.md` for now. Should I change that to `09_enums.md`?
Also, I am not sure on whether the id in the link for switch expressions is correct since that article seems to be missing from this repository as well.

### CODE REVIEW

File: app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md
Comment (wkorando) [inline app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md:41]: Update section title to Accessing, evaluating, and comparing enums.

Add an example of using "compareTo", there are some natural examples like: 

```java
public void compareDayOfWeek(DayOfWeek dayOfWeek){
   int comparison = dayOfWeek.compareTo(DayOfWeek.WEDNESDAY);
   if ( comparison == -1) {
      System.out.println("It's before the middle of the work week.");
   } else if(comparison == 0){
      System.out.println("It's the middle of the work week.");
   } else {
      System.out.println("It's after the middle of the work week.");
   }
}
```

Of course, feel free to change that example as desired, and in the description, mention the declaration order of the enum instances in the enum class is the order that will be used for comparison.

File: app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md
Comment (wkorando) [inline app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md:168]: Some kind of summary should be good here as well. As this ends rather abruptly. Perhaps some discussion regarding when enums should and shouldn't be used? e.g. using an enum in a shared library for something that might change in the future could create a lot of headaches. Of course, if you have a better example, or way of wrapping up enums that's fine as well.

**Comment by wkorando** [comment]
Thank you for the contribution Dan! I made some comments on areas for improvement.

File: app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md
Comment (danthe1st) [reply app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md:92]: I wrote `the compiler can check whether all values of the enum are handled` and I originally didn't add more as this article was about enums and not switch expression but I can add this.

File: app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md
Comment (danthe1st) [reply app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md:41]: I think it makes more sense to put this in the section `Special methods`.

File: app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md
Comment (wkorando) [reply app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md:92]: Agreed, though just for clarification, I was referring to a switch **statement** that uses pattern matching would also need to be exhaustive. As it stands now, the wording implies the exhaustiveness check only applies to expressions which is no longer true as of Java 21.

File: app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md
Comment (danthe1st) [reply app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md]: I have now added it there. If you think it should still be put in `Accessing and comparing enums`, let me know.

**Comment by danthe1st** [comment]
Can someone please check whether the directory names, slug/id and similar as well as the link to Switch Expressions are correct? I had to guess these since these articles/the `Classes and objects` section are missing in this repository.

**Comment by danthe1st** [comment]
And I can also squash the changes to a single commit if wanted.

File: app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md
Comment (ammbra) [inline app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md:192]: In my opinion, the conclusion is tough to understand for a beginner. In general enums are used to represent a fixed set of constants and I think would be great to have that as a takeaway.

File: app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md
Comment (ammbra) [inline app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md:195]: this paragraph makes me believe that whenever I change an enum constant I made end with surprising errors; maybe you would like to warn the reader that refactoring enum constants should be performed after analysing their code usage across files?

File: app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md
Comment (ammbra) [inline app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md:200]: This phrase is enigmatic: there are other options when having a big number of constants in the same file. Some people would wonder how many constants is too many for an enum? and what are the other options?

File: app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md
Comment (wkorando) [reply app/pages/learn/01_tutorial/03_getting-to-know-the-language/04_classes_objects/01_enums.md]: Yea, placing it under special methods makes sense, especially with the additional explanation you provided.

**Comment by wkorando** [comment]
> And I can also squash the changes to a single commit if wanted.

Yea if you can do that at the end. That makes reviewing this addition later easier.

**Comment by danthe1st** [comment]
I have now made a few changes to the conclusion section in order to address @ammbra's feedback and make it easier to understand.
I am thinking about removing the last paragraph altogether as it might be a bit confusing and ambiguous as already mentioned.

**Comment by carimura** [comment]
Moving to PR #47 since I worked in a different branch. Sorry! Once comments are resolved here, we can update #47 and merge.

**Comment by danthe1st** [comment]
I just want to mention that with this approach, I will not be able to squash it or do any changes in that branch so one of you would need to do this if needed.

**Comment by carimura** [comment]
no problem you can work in your branch and I can update. I've changed the conclusion a bit, see if you like it.

**Comment by carimura** [comment]
Can you merge the changes from https://github.com/java/devjava-content/compare/main...danthe1st-enums into your branch so this PR is the active one?

**Comment by danthe1st** [comment]
I have merged it as well as changed `another` to `a` since the article series doesn't mention other types of classes.

**Review by carimura — APPROVED** [review]

**Comment by carimura** [comment]
Thank you @danthe1st for your contribution and congrats on being the first approved community contribution! Once merged here we'll port it to the live site soon. Shoot me an email or Twitter DM and we'll arrange for a small thank you to be sent your way.

(thanks @wkorando and @ammbra for your reviews)

**Comment by carimura** [comment]
in case you missed it https://dev.java/learn/classes-objects/enums/

**Pull request merged**
